### PR TITLE
Credential add/edit component

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- visit /account
- check you can still add and edit credentials

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8454/

**Comment by huwshimi** [comment]
[Credential add/edit component](https://trello.com/c/qE12qLfs/264-credential-add-edit-component)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8455/

File: jujugui/static/gui/src/app/components/credential-add-edit/credential-add-edit.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+const DeploymentCloud = require('../deployment-flow/cloud/cloud');
+const DeploymentCredentialAdd = require('../deployment-flow/credential/add/add');
+const GenericButton = require('../generic-button/generic-button');
+const Spinner = require('../spinner/spinner');
+
+// List, add and remove cloud credentials in the account page.
+class CredentialAddEdit extends React.Component {
+  constructor() {
+    super();
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/credential-add-edit/credential-add-edit.js]: pass the props through?

File: jujugui/static/gui/src/app/components/credential-add-edit/credential-add-edit.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+      xhr && xhr.abort && xhr.abort();
+    });
+  }
+
+  /**
+    Retrieve the list of clouds.
+  */
+  _getClouds() {
+    this.setState({loading: true}, () => {
+      const xhr = this.props.controllerAPI.listClouds((error, clouds) => {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/credential-add-edit/credential-add-edit.js]: You can remove all the xhr bits as this is not an xhr call

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8458/

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Comment by huwshimi** [comment]
[Credential add/edit component](https://trello.com/c/qE12qLfs/264-credential-add-edit-component)

**Pull request merged**